components/src/DateRange/DateRange.js

```diff
@@ -0,0 +1,160 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import { isBefore, addDays, differenceInDays } from "date-fns";
+import { DatePicker } from "../DatePicker";
+import { RangeContainer } from "../RangeContainer";
+import { InputFieldPropTypes, InputFieldDefaultProps } from "../Input/InputField.type";
+import { FieldLabelDefaultProps, FieldLabelProps } from "../FieldLabel/FieldLabel.type";
+import { DateRangeStyles } from "./DateRangeStyles";
+
+const DateRange = ({
+  dateFormat,
+  onRangeChange,
+  onStartDateChange,
+  onEndDateChange,
+  errorMessage,
+  startDateErrorMessage,
+  endDateErrorMessage,
+  defaultStartDate,
+  defaultEndDate,
+  endDateInputProps,
+  startDateInputProps,
+  disableRangeValidation,
+  labelProps,
+  minDate,
+  maxDate
+}) => {
+  const [startDate, setStartDate] = useState(defaultStartDate);
+  const [endDate, setEndDate] = useState(defaultEndDate);
+  const [rangeError, setRangeError] = useState();
+
+  const changeStartDateHandler = date => {
+    setStartDate(date);
+    if (onStartDateChange) {
+      onStartDateChange(date);
+    }
+  };
+  const changeEndDateHandler = date => {
+    setEndDate(date);
+    if (onEndDateChange) {
+      onEndDateChange(date);
+    }
+  };
+
+  const validateDateRange = () => {
+    let error;
+    if (endDate && startDate && isBefore(endDate, startDate)) {
+      error = "End date is before start date";
+    }
+    setRangeError(error);
+    if (onRangeChange) {
+      onRangeChange({
+        startDate,
+        endDate,
+        error
+      });
+    }
+  };
+
+  const getAllDaysInRange = () => {
+    if (endDate && startDate && isBefore(startDate, endDate)) {
+      const days = Array(differenceInDays(new Date(endDate), new Date(startDate)) + 1);
+      return days.fill(0).map((_, i) => addDays(new Date(startDate), i));
+    }
+    return [];
+  };
+
+  const highlightDates = [
```

Comment:
should this be in the styles file?

Reply:
Yes actually that would be nice, moving it